src/test/scala/uk/gov/hmrc/api/utils/BaseSpec.scala

```diff
@@ -46,12 +46,14 @@ trait BaseSpec extends AnyFeatureSpec with GivenWhenThen with Matchers with Befo
   override def beforeAll(): Unit = {
     super.beforeAll()
 
-    val thirdPartyApplicationResponse =
-      ppnsService.createClientApplication(thirdpartyApplicationHadersMap)
-    thirdPartyApplicationResponse.status shouldBe 201
+    withClue("Setup step failed: Create Client Application → ") {
+      val thirdPartyApplicationResponse =
+        ppnsService.createClientApplication(thirdpartyApplicationHadersMap)
+      thirdPartyApplicationResponse.status should (be(201) or be(200))
 
-    val json = Json.parse(thirdPartyApplicationResponse.body)
-    clientId = (json \ "details" \ "token" \ "clientId").as[String]
+      val json = Json.parse(thirdPartyApplicationResponse.body)
+      clientId = (json \ "details" \ "token" \ "clientId").as[String]
```

Comment:
I suppose if this ever changed again :shrug:  Maybe somnething like:
```
json.validate[String]((__ \\ "clientId").read[String]) match {
      case JsSuccess(value, _) => value
      case JsError(errors)     => throw new NoSuchElementException(errors.toString())
    }
    ```

Reply:
Thanks Tom, appreciate it. I think we were on the same wave length. I've fixed this as part of my refactoring comment above, although not throwing any exceptions, just failures that are reported somewhat gracefully for the spec in question👍🏼